lib/Extension/LanguageServerCodeTransform/LspCommand/GenerateDecoratorCommand.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerCodeTransform\LspCommand;
+
+use Amp\Promise;
+use Amp\Success;
+use Phpactor\CodeTransform\Domain\Exception\TransformException;
+use Phpactor\Extension\LanguageServerBridge\Converter\TextEditConverter;
+use Phpactor\LanguageServerProtocol\Command;
+use Phpactor\LanguageServerProtocol\WorkspaceEdit;
+use Phpactor\LanguageServer\Core\Server\ClientApi;
+use Phpactor\CodeTransform\Domain\SourceCode;
+
+class GenerateDecoratorCommand extends Command
+{
+    public const NAME = 'generate_decorator';
+
+    private ClientApi $clientApi;
+
+    public function __construct(ClientApi $clientApi)
+    {
+        $this->clientApi = $clientApi;
+    }
+
+    /**
+     * @return Promise<ApplyWorkspaceEditResponse>
+     */
+    public function __invoke(string $uri): Promise
```

Comment:
I would possibly have the command accept an interface name `$interfaceFqn` - that way it's responsiblity may be reduced and we have a clearer distinction between the code action provider and the command.

Reply:
Done that.